server/manager/src/main/java/org/apache/accumulo/manager/tableOps/compact/CompactRange.java

```diff
@@ -83,7 +92,36 @@ public long isReady(FateId fateId, Manager env) throws Exception {
   @Override
   public Repo<Manager> call(final FateId fateId, Manager env) throws Exception {
     CompactionConfigStorage.setConfig(env.getContext(), fateId, config);
-    return new CompactionDriver(namespaceId, tableId, startRow, endRow);
+    KeyExtent keyExtent;
+    byte[] prevRowOfStartRowTablet = startRow;
+    byte[] endRowOfEndRowTablet = endRow;
+
+    if (startRow != null) {
+      // The startRow in a compaction range is not inclusive, so do not want to find the tablet
+      // containing startRow but instead find the tablet that contains the next possible row after
+      // startRow
+      Text nextPossibleRow = new Key(startRow).followingKey(PartialKey.ROW).getRow();
+      keyExtent = findContaining(env.getContext().getAmple(), tableId, nextPossibleRow);
+      prevRowOfStartRowTablet = keyExtent.prevEndRow().getBytes();
```

Comment:
The found extent could have a null prev end row.  Also the `getBytes()` method on Text can be dangerous because it can return an array that is longer than the data.  The Accumulo `TextUtil.getBytes()` method covers this.  We probably need a util method that covers the entire pattern below.

```suggestion
      prevRowOfStartRowTablet = keyExtent.prevEndRow() == null ? null : TextUtil.getBytes(keyExtent.prevEndRow().getBytes());
```